category.go

```diff
@@ -1,49 +1,56 @@
 package cli
 
-// CommandCategories is a slice of *CommandCategory.
+// CommandCategories wraps a slice of *CommandCategory.
 type CommandCategories struct {
```

Comment:
Doh, I missed that this mimics the existing interface. Though now that we are hiding the slice, it might make sense to expose methods for removing elements :smile: